### PR TITLE
Replace CustomPoint with extension methods on Point

### PR DESCRIPTION
**Pull request by rorystephenson**
The CustomPoint class predates dart extension methods. The name is confusing (custom in what sense, and why) and in looking at the implementation it became clear that it was essentially just adding extension methods.

Changing to a Point will avoid conversions in plugins which deal with libraries that use Point.

Code will work as-is without changes with the exception of the subtraction operator. Previously CustomPoint<int> - CustomPoint<double> did not trigger analyzer warnings but would cause a runtime exception. This is now replaced by subtract() extension methods which allow subtracting num from a Point<double> and int from Point<int>.

---

Issue #1522 prompted me to look at this and I believe it resolves that issue (we no longer add methods which are not runtime-safe to Point and Point already documents the potential runtime error with the * operator).

Note that the vast majority of changes are just renaming CustomPoint to Point and importing it.

### CODE REVIEW

File: lib/src/geo/crs.dart [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import 'dart:math' as math;
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/geo/crs.dart:2]: Use only one import for the same library.

File: lib/src/layer/tile_layer/tile_layer.dart [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import 'dart:async';
 import 'dart:math' as math;
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/layer/tile_layer/tile_layer.dart:3]: Use only one import for the same library.

File: lib/src/layer/tile_layer/tile_range.dart [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import 'dart:math' as math;
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/layer/tile_layer/tile_range.dart:2]: Use only one import for the same library.

File: lib/src/map/camera/camera.dart [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import 'dart:math' as math;
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/map/camera/camera.dart:2]: Use only one import for the same library.

File: lib/src/map/camera/camera_constraint.dart [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import 'dart:math' as math;
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/map/camera/camera_constraint.dart:2]: Use only one import for the same library.

File: lib/src/map/camera/camera_fit.dart [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import 'dart:math' as math;
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/map/camera/camera_fit.dart:2]: Use only one import for the same library.

File: lib/src/misc/point_extensions.dart [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  /// values in [other].
+  Point<int> scaleBy(Point<int> other) {
+    return Point<int>(x * other.x, y * other.y);
+  }
+}
+
+extension OffsetToPointExtension on Offset {
+  @Deprecated(
+    'Prefer `toPoint()`. '
+    'This method has been renamed as a result of CustomPoint\'s removal. '
```
Comment (JaffaKetchup) [inline lib/src/misc/point_extensions.dart:121]: Just for consistency, prefer double quotes with single inner quotes over single quotes with escaped inner quotes.

```suggestion
    "This method has been renamed as a result of CustomPoint's removal. "
```

File: lib/src/misc/private/bounds.dart [hunk 8]
```diff
@@ -1,22 +1,21 @@
 import 'dart:math' as math;
-
-import 'package:flutter_map/src/misc/point.dart';
+import 'dart:math';
```
Comment (JaffaKetchup) [inline lib/src/misc/private/bounds.dart:2]: Use only one import for the same library.

**Review by JaffaKetchup — CHANGES_REQUESTED** [review]
LGTM, apart from these few nitpicks for consistency and to keep things concise.

File: lib/src/geo/crs.dart (hunk 1 above)
Comment (rorystephenson) [reply lib/src/geo/crs.dart:2]: Nice catch 👍

**Comment by rorystephenson** [comment]
Nice catch on the redundant imports @JaffaKetchup !

First I started by removing the `import 'dart:math' as math;` imports but that didn't work well with some classes which had `min`/`max` getters in classes in some of the tile related code. So then I started removing the new `import 'dart:math';` imports but this meant that `Point` was always prefixed with `math` which makes quite a mess. I think importing math with a prefix makes sense for methods like `min`/`max` and getters like `pi` but in the case of `Point` it is pretty unambiguous.

For that reason I've updated the imports to:

```dart
import 'dart:math' as math hide Point;
import 'dart:math' show Point;
```

WDYT?

**Comment by JaffaKetchup** [comment]
It's less than ideal, I'm sure you'll agree, but OK for now!

**Review by mootw — APPROVED** [review]
lgtm

**Pull request merged**